core/shared/src/main/scala/cats/parse/Accumulator.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2020 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats.parse
+
+import cats.data.{NonEmptyList, NonEmptyVector}
+import scala.collection.mutable.Builder
+
+trait Appender[-A, +B] {
+  def append(item: A): this.type
+  def finish(): B
+}
+
+object Appender {
+  def charStringAppender(): Appender[Char, String] =
+    new Appender[Char, String] {
+      val bldr = new java.lang.StringBuilder()
+
+      def append(item: Char) = {
+        bldr.append(item)
+        this
+      }
+
+      def finish(): String = bldr.toString()
+    }
+
+  def stringAppender(): Appender[String, String] =
+    new Appender[String, String] {
+      val bldr = new java.lang.StringBuilder()
+
+      def append(item: String) = {
+        bldr.append(item)
+        this
+      }
+
+      def finish(): String = bldr.toString()
+    }
+
+  def fromBuilder[A, B](bldr: Builder[A, B]): Appender[A, B] =
+    new Appender[A, B] {
+      def append(item: A) = {
+        bldr += item
+        this
+      }
+
+      def finish(): B = bldr.result()
+    }
+}
+
+trait Accumulator[-A, +B] {
```

Comment:
It seems like having `Accumulator` extend `Accumulator1` would make sense, since:

```scala
def newAppender(first: A): Appender[A, B] = {
  val appender = newAppender()
  appender.append(first)
  appender
}
```

Reply:
good call. I did this.